update function names and zero_point checks

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59701**

Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

HDCharles committed:
update internal function names that apply to both cpu and cuda

Summary: quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "update internal function names that apply to both cpu and cuda"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update internal function names that apply to both cpu and cuda"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

vkuzo commented on aten/src/ATen/native/quantized/cuda/affine_quantizer.cu:
```diff
@@ -47,39 +62,37 @@ void dequantize_tensor_per_tensor_affine_cuda(
+    const Tensor& scales,
+    const Tensor& zero_points,
+    int64_t axis) {
+  static const std::string fn_name = "quantize_tensor_per_channel_affine_cuda";
```
what does this do?

HDCharles replied:
something weird happened to this PR, only like half the changes are here, this change was added because we use it both for the macro dispatch and the zero_point checks.

HDCharles replied:
ok, its fixed now

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

supriyar commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -5114,13 +5114,13 @@
   variants: function
   dispatch:
-    CPU, CUDA: quantize_per_channel_cpu
+    CPU, CUDA: quantize_per_channel
```
Are there any BC concerns with re-naming this operator?

supriyar replied:
Not sure if we have any use-cases where we directly call `at::native::quantize_per_channel_cpu`

HDCharles replied:
I was under the impression that anything external would just go through the dispatcher.

HDCharles replied:
perhaps @raghuramank100 or @vkuzo would have a better idea

raghuramank100 replied:
Doing fbgs shows only internal usage within torch : https://www.internalfb.com/intern/codesearch/?bunny_arg&bunny_command=fbgs&lucky=0&q=repo%3Afbcode%20case%3Ainsensitive%20regex%3Aoff%20quantize_per_channel_cpu&source=redirect
so this is ok

supriyar commented on aten/src/ATen/native/quantized/cuda/affine_quantizer.cu:
```diff
@@ -1,15 +1,29 @@
+  TORCH_CHECK(
+    zp_within_upper and zp_within_lower,
+    fn_name,
+    " zero_point is out of range.");
```
Maybe adding information of what is out of range would be useful (lower/upper)

HDCharles replied:
I'm not sure, I was under the impression that anything external would go through the dispatcher

HDCharles replied:
good idea, fixed

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

raghuramank100 reviewed (approved):
Looks good to me

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles committed:
Update on "update function names and zero_point checks"


Summary: 
These functions have been updated to work for cpu and cuda, their names are now changed to reflect that

quantize_per_channel_cpu -> quantize_per_channel
dequantize_quantized_cpu -> dequantize_quantized

the zero point checks were bypassed for cuda tensors, this fixes that

Test Plan: python test/test_quantization.py TestQuantizedTensor

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29018270](https://our.internmc.facebook.com/intern/diff/D29018270)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29018270).
